tests/baselines/reference/conditionalExportsResolutionFallback(moduleresolution=bundler).errors.txt

```diff
@@ -1,7 +1,9 @@
 error TS5095: Option 'bundler' can only be used when 'module' is set to 'es2015' or later.
+error TS5109: Option 'moduleResolution' must be set to 'NodeNext' (or left unspecified) when option 'module' is set to 'NodeNext'.
 
 
 !!! error TS5095: Option 'bundler' can only be used when 'module' is set to 'es2015' or later.

```

Comment:
(Since this format data directly controls, eg, how we interpret `default` imports, the distinction may be material)

Reply:
Yeah, that's the linked issue #54102 - shouldn't we just fix that, while we're here? It should be removing this error, right?

Reply:
Yeah, that is the ultimate goal, but I don’t think `--module nodenext` is the right thing to combine with `--moduleResolution bundler`. `--module nodenext` implies that CJS-mode files actually get emitted as CJS, which doesn’t make sense in the context of most bundlers. In bundlers, the choice is not between CJS and ESM; the default is ESM with Babel-like interop and `.mjs` files just disable the Babel-like interop in favor of Node’s always-synthesize-a-default interop. It can be hard to tell the difference between “ESM with Babel-like interop” and “it’s actually just CJS with `esModuleInterop` under the hood,” but we can tell there’s a difference because

- top-level wait is always allowed (by those that support it at all)
- the module format does not affect the `"imports"`/`"exports"` conditions used in module resolution—import statements always have `"import"`; they’re not secretly transformed to `require` first in CJS-mode files
- some bundlers refuse to process `require` calls outside of node_modules at all

Additionally, the transform for `import foo = require(...)` to a `createRequire` call under `--module nodenext` is highly Node-specific and would fail under I think every bundler (maybe Webpack has compat for that; they do the most complete job at imitating Node).

So, my thinking for addressing #54102 is that either there needs to be a different `module` mode for these bundlers that have Node-like interop behavior (in which case this PR is necessary but not sufficient), or the thing that makes us set `impliedNodeFormat` in the first place needs to be decoupled from `module` (in which case this PR is not necessary, but doesn’t hurt anything as an interim step toward the eventual goal). Either way, we’ll need to move it off of `moduleResolution` in order to resolve #54102, and `module` will _imply_ an `impliedNodeFormat`-setting behavior if it doesn’t directly control it.